### ISSUE TITLE
[OJS 3.x] Notify user when email sending fails?

### ISSUE THREAD

**Issue by ctgraham** [pkp/pkp-lib#2422]
`Mail::send` returns true/false depending on success or failure of sending mail, but the caller is pretty much always constructed as: [`$mail->send();`](https://github.com/pkp/pkp-lib/blob/91da80b797e2f7dd98014134251c2884de6f30c5/controllers/grid/settings/user/form/UserEmailForm.inc.php#L81-L86) without any check of the return value.  Many? most? all? of these should trigger a notification via the UI regarding a failed email initiated by the UI.  Certainly any form-based sending, probably any workflow based sending, perhaps others?

**Comment by asmecher** [comment]
Agreed -- OJS 2.x was very inconsistent/spotty around error handling, and OJS 3.x still has some of that legacy. Now that we've entered the 21st century (Exceptions etc.) we could probably stand to have a discussion about error handling more generally. For example, when to just report back to the user (e.g. CSRF token mismatch), when to log an error (email sending failure), and in the latter case, how much information to provide to users; also some general coding practices (e.g. use of exceptions).

**Comment by ajnyga** [comment]
We had a related issue where a clear error message would had helped. 

An editor was submitting an editorial decision and attached a reviewer file to the message. The file however was over 16 megabytes and when the decisions was made and OJS tried to send the email to the author it failed.  The reason was probably the default value (10 megabytes) of Postifx attachment max size setting (not 100% sure yet, will update here when I am sure)

The error log had "Could not instantiate mail function" errors, but there was no error visible for the editor. I think this could/should be easily fixed?

**Comment by asmecher** [comment]
PRs:
- https://github.com/pkp/pkp-lib/pull/4237
- https://github.com/pkp/ojs/pull/2168
- https://github.com/pkp/omp/pull/604

Note that notifications have not been added where the task is potentially only server-side (e.g. cron job).

**Comment by asmecher** [comment]
Merged for OJS 3.2.
